modules/subdyn/src/SubDyn.f90

```diff
@@ -1152,13 +1152,45 @@ SUBROUTINE SD_Input(SDInputFile, Init, p, ErrStat,ErrMsg)
 if (LegacyFormat) then
    nColumns = 5
    Init%Members(:,iMType) = idMemberBeam ! Important, in legacy all members are beams
+   Init%Members(:,7) = -1
 else
    nColumns = MembersCol
 endif
-DO I = 1, p%NMembers
+
+nColumns=MembersCol
+CALL AllocAry(StrArray, nColumns, 'StrArray',ErrStat2,ErrMsg2); if (Failed()) return 
+READ(UnIn, FMT='(A)', IOSTAT=ErrStat2) Line  ; ErrMsg2='First line of members array'; if (Failed()) return
+CALL ReadCAryFromStr ( Line, StrArray, nColumns, 'Members', 'First line of members array', ErrStat2, ErrMsg2 )
+if (ErrStat2/=0) then
+   ! We try with 4 columns (legacy format)
+   nColumns = MembersCol-1
+   deallocate(StrArray)
+   CALL AllocAry(StrArray, nColumns, 'StrArray',ErrStat2,ErrMsg2); if (Failed()) return 
+   CALL ReadCAryFromStr ( Line, StrArray, nColumns, 'Members', 'First line of members array', ErrStat2, ErrMsg2 ); if(Failed()) return
+   call LegacyWarning('Member table contains 6 columns instead of 7, using default member cosines.')
```

Comment:
Suggest modifying to: ", using default member directional cosines ID (-1) for all members. The directional cosine will be computed based on the member nodes for all members. "